### PR TITLE
Added configuration option to disable auditing for object associations

### LINKED ISSUE #601: Ignore related tables for auditiong
## Feature Request
I would like to ignore a related table with M:M for auditing.

Actually it is not possible as I see and I cannot change the current library.  

https://stackoverflow.com/questions/77801195/how-to-ignore-many-to-many-relationship-with-entityaudit-entityauditbundle?noredirect=1#comment137179292_77801195 

```php
<?php
/** @Entity **/
class User
{
    // ...

    /**
     * @ManyToMany(targetEntity="User", mappedBy="myFriends")
     **/
    private $friendsWithMe;

    /**
     * @ManyToMany(targetEntity="User", inversedBy="friendsWithMe")
     * @JoinTable(name="friends",
     *      joinColumns={@JoinColumn(name="user_id", referencedColumnName="id")},
     *      inverseJoinColumns={@JoinColumn(name="friend_user_id", referencedColumnName="id")}
     *      )
     **/
    private $myFriends;

    public function __construct() {
        $this->friendsWithMe = new \Doctrine\Common\Collections\ArrayCollection();
        $this->myFriends = new \Doctrine\Common\Collections\ArrayCollection();
    }

    // ...
}
```

```
simple_things_entity_audit:
    audited_entities:
        - App\Entity\User
        - 
    global_ignore_tables:
        - friends
```

Thanks.

### PR DESCRIPTION
**Pull request by shepherdmat**
## Subject

Pull request allows you to globally disable building a database scheme, and attempt to audit object relationships

I am targeting this branch, because it is a new feature.

Closes #601.

## Changelog

```markdown
### Added
- New configuration option `disable_associations` to disable association tracking.
```

### CODE REVIEW

File: src/EventListener/LogRevisionsListener.php [hunk 1]
```diff
@@ -608,13 +612,18 @@ private function saveRevisionEntityData(EntityManagerInterface $em, ClassMetadat
     }
 
     /**
-     * @param array<string, mixed>  $assoc
-     * @param array<string, mixed>  $entityData
+     * @param array<string, mixed> $assoc
+     * @param array<string, mixed> $entityData
```
Comment (VincentLanglet) [inline src/EventListener/LogRevisionsListener.php]: You'll get a cs failure from this change

File: src/EventListener/LogRevisionsListener.php [hunk 2]
```diff
@@ -608,13 +612,18 @@ private function saveRevisionEntityData(EntityManagerInterface $em, ClassMetadat
...
      * @param ClassMetadata<object> $class
      * @param ClassMetadata<object> $targetClass
+     * @throws Exception
      */
     private function recordRevisionForManyToManyEntity(object $relatedEntity, EntityManagerInterface $em, string $revType, array $entityData, array $assoc, ClassMetadata $class, ClassMetadata $targetClass): void
     {
+        if ($this->config->areAssociationsDisabled()) {
+            throw new \RuntimeException('This method is unreachable because option "disable_associations" is enabled.');
+        }
```
Comment (VincentLanglet) [inline src/EventListener/LogRevisionsListener.php]: This exception does not seem needed to me. You can remove it.

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

File: src/EventListener/LogRevisionsListener.php (hunk 1 above)
Comment (shepherdmat) [reply src/EventListener/LogRevisionsListener.php]: I have made the indicated changes in the latest commit

File: src/EventListener/LogRevisionsListener.php (hunk 2 above)
Comment (shepherdmat) [reply src/EventListener/LogRevisionsListener.php]: I have made the indicated changes in the latest commit

File: src/Collection/AuditedCollection.php [hunk 3]
```diff
@@ -157,14 +157,12 @@ public function containsKey($key)
     }
 
     /**
-     * @param string|int $key
```
Comment (VincentLanglet) [inline src/Collection/AuditedCollection.php]: Please revert this change

File: src/Collection/AuditedCollection.php [hunk 4]
```diff
@@ -157,14 +157,12 @@ public function containsKey($key)
...
-     *
      * @return object
      *
      * @phpstan-return T
      */
     #[\ReturnTypeWillChange]
-    public function get($key)
+    public function get(mixed $key)
```
Comment (VincentLanglet) [inline src/Collection/AuditedCollection.php]: Please revert this change

**Comment by VincentLanglet** [comment]
You also need to rebase 1.x branch @shepherdmat

File: src/Collection/AuditedCollection.php (hunk 3 above)
Comment (shepherdmat) [reply src/Collection/AuditedCollection.php]: Done

File: src/Collection/AuditedCollection.php (hunk 4 above)
Comment (shepherdmat) [reply src/Collection/AuditedCollection.php]: Done

**Comment by shepherdmat** [comment]
@VincentLanglet Sorry. I made a mess in git had to create new PR here  https://github.com/sonata-project/EntityAuditBundle/pull/622 Thank you in advance for review
